std/root.ln

```diff
@@ -232,6 +569,7 @@ export fn xnor(a: int64, b: int64) = xnori64(a, b);
 export fn xnor(a: bool, b: bool) = xnorboo(a, b);
 
 // Equality and order functions
+// TODO: Similarly, should equality/orderability functions accept Result-wrapped values?
```

Comment:
Maybe, but then we get into a slippery slope of implicit type conversions and seemingly impossible comparison results.

Possibilities here:
1. Keep it as-is -- the user has to manually unwrap the result-wrapped integer to do a comparison, eg `if j < (i - 1 || 0) ...` This adds syntactic noise but is clearer.
2. Result-wrap the comparison if Result-wrapped integers are involved, but this requires unwrapping the final comparison `if j < i - 1 || false ...` This makes it clearer that on failure you're explicitly choosing the success/failure of the conditional, but it still looks really weird.
3. Comparisons with result-wrapped integers always return an unwrapped boolean `if j < i - 1 ...` looks like before this PR, but has implicit rules to memorize, like "if either comparison input is an error type, returns false" and can cause weird things like `if j < i - 1 ... else if i - 1 < j ...` both being `false`.

The third behavior is most like how Javascript does things. The first behavior is most like Rust, while the middle is an "infectious" style kinda like Java exception bubbling. (The error opcode is simply passed up to something else that cares about it.)

I am personally leaning towards (3) as the code looks the closest to other popular languages and the comparison rules are also similar:

```
damocles@Talyn:/usr/local/alancorp/alan(resultify-math)$ node
> 3 < undefined
false
> undefined < 3
false
>
```
But we didn't discuss this issue in the RFC and so it would be nice for some thoughts from others.

Reply:
I much prefer the third option to the rest so that sounds good!